tests/PHPStan/Rules/Comparison/StrictComparisonOfDifferentTypesRuleTest.php

```diff
@@ -882,4 +882,14 @@ public function testBug8776Part2(): void
 		$this->analyse([__DIR__ . '/data/bug-8776-2.php'], []);
 	}
 
+	public function testBug5978(): void
+	{
+		if (PHP_VERSION_ID >= 80000) {
+			$this->markTestSkipped('Test does not run on PHP >= 8.0.');
```

Comment:
Instead, you could assert different errors on PHP 7.x vs. 8.x.

Reply:
Indeed, that's better